### PR TITLE
Use cache to retrieve WorkflowTemplates

### PR DESCRIPTION
**Pull request by dtaniwaki**
Solves #1529 

I used an informer to efficiently query WorkflowTemplates.

### CODE REVIEW

File: workflow/controller/controller.go [hunk 1]
```diff
@@ -145,16 +149,20 @@ func (wfc *WorkflowController) Run(ctx context.Context, wfWorkers, podWorkers in
 
 	wfc.wfInformer = util.NewWorkflowInformer(wfc.restConfig, wfc.Config.Namespace, workflowResyncPeriod, wfc.tweakWorkflowlist)
 
+	informerFactory := wfextv.NewSharedInformerFactory(wfc.wfclientset, workflowTemplateResyncPeriod)
+	wfc.wftmplInformer = informerFactory.Argoproj().V1alpha1().WorkflowTemplates()
+
 	wfc.addWorkflowInformerHandler()
 	wfc.podInformer = wfc.newPodInformer()
 
 	go wfc.wfInformer.Run(ctx.Done())
```
Comment (xianlubird) [inline workflow/controller/controller.go]: Will the old wfInformer also watch and get wf info from apiServer ?

Comment (dtaniwaki) [reply workflow/controller/controller.go]: I think `wfInformer` is only for `Workflows`.
https://github.com/argoproj/argo/blob/master/workflow/util/util.go#L57

Comment (xianlubird) [reply workflow/controller/controller.go]: Yep,   LGTM

**Comment by gpomykala** [comment]
same fix makes much sense in argo clients, it was raised up back in 2020, is there any plan to enable client-side caching as well? https://github.com/argoproj/argo-workflows/issues/3899

**Pull request merged**